modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -10066,6 +10079,25 @@ private void doProcessCSS() {
         }
     }
 
+    /**
+     * A node is considered to be in its initial CSS state if it wasn't shown in a scene graph before.
+     * This flag is cleared after CSS processing was completed in a Scene pulse event. Note that manual
+     * calls to {@link #applyCss()} or similar methods will not clear this flag, since we consider all
+     * CSS processing before the Scene pulse to be part of the node's initial state.
+     */
+    private boolean initialCssState = true;
```

Comment:
is it possible to find a solution that does not involve adding a field to Node?

Reply:
I don't think so, because we need to track the "initial CSS state" flag for each individual node. It can change during runtime when the node is added to or removed from the scene graph.

I've looked at adding it to `CssFlags` or `DirtyBits`, but both have different purposes. The addition of this boolean field doesn't change the object size for most HotSpot implementations, as boolean fields are usually packed together. Currently, `Node` has 9 boolean fields (plus 1 byte field), with an extra 2 bytes of unused space.